finality-aleph/src/party.rs

```diff
@@ -310,24 +310,20 @@ async fn run_session_as_authority<B, C, BE, SC>(
         }
     };
 
-    spawn_handle
-        .0
-        .spawn("aleph/consensus_session_member", member_task);
-    spawn_handle
-        .0
-        .spawn("aleph/consensus_session_forwarder", forwarder_task);
-    spawn_handle
-        .0
-        .spawn("aleph/consensus_session_aggregator", aggregator_task);
+    let member_handle = spawn_handle.spawn_essential("aleph/consensus_session_member", member_task);
+    let forwarder_handle =
+        spawn_handle.spawn_essential("aleph/consensus_session_forwarder", forwarder_task);
+    let aggregator_handle =
+        spawn_handle.spawn_essential("aleph/consensus_session_aggregator", aggregator_task);
 
     let _ = exit_rx.await;
     info!(target: "afa", "Shutting down authority session {}", session_id.0);
     let _ = exit_member_tx.send(());
-    debug!(target: "afa", "Waiting 5000ms for Member to shut down without panic");
-    Delay::new(Duration::from_millis(5000)).await;
-    // This is a temporary solution -- need to fix this in AlephBFT.
+    let _ = member_handle.await;
     let _ = exit_aggregator_tx.send(());
+    let _ = aggregator_handle.await;
     let _ = exit_forwarder_tx.send(());
+    let _ = forwarder_handle.await;
```

Comment:
This indeed should work, but it is a little unsatisfying that the ordering matters here :/  Maybe add some comment saying that this order is correct and why...

Reply:
I've changed the order in which they are spawned too. Worker tasks are spawned before forwarder now and forwarder is "dropped" last to avoid those "send on closed channel" panics. I've added some comment regarding that.